metrics/exporter.go

```diff
@@ -215,24 +199,20 @@ func newMetricsExporter(config *metricsConfig, logger *zap.SugaredLogger) (view.
 		se.StopMetricsExporter()
 	}
 
-	var err error
-	var e view.Exporter
-	switch config.backendDestination {
-	case openCensus:
-		e, err = newOpenCensusExporter(config, logger)
-	case stackdriver:
-		e, err = newStackdriverExporter(config, logger)
-	case prometheus:
-		e, err = newPrometheusExporter(config, logger)
-	case none:
-		e, err = nil, nil
-	default:
-		err = fmt.Errorf("unsupported metrics backend %v", config.backendDestination)
+	factory := map[metricsBackend]func(*metricsConfig, *zap.SugaredLogger) (view.Exporter, ResourceExporterFactory, error){
+		stackdriver: newStackdriverExporter,
+		openCensus:  newOpenCensusExporter,
+		prometheus:  newPrometheusExporter,
+		none: func(*metricsConfig, *zap.SugaredLogger) (view.Exporter, ResourceExporterFactory, error) {
+			return nil, nil, nil
+		},
```

Comment:
I have a slight preference (since it is only used one place, and infrequently) to keep this close to the code that uses it, but I don't feel strongly.